walkeeper/src/timeline.rs

```diff
@@ -156,6 +158,30 @@ impl SharedState {
         })
     }
 
+    // Check the magic/version in the on-disk data and deserialize it, if possible
+    fn deser_sk_state(buf: &mut &[u8], control_file_path: &Path) -> Result<SafeKeeperState> {
```

Comment:
`control_file_path` is used only for error messages, can be applied at the caller side instead of passing as argument